Code/Framework/AzFramework/AzFramework/API/ApplicationAPI.h

```diff
@@ -189,18 +189,32 @@ namespace AzFramework
         virtual void OnApplicationAboutToStop() {}
     };
 
-    class LevelSystemLifecycleRequests
+    class LevelLoadBlockerRequests
         : public AZ::EBusTraits
     {
     public:
-        // Bus Configuration
         static const AZ::EBusHandlerPolicy HandlerPolicy = AZ::EBusHandlerPolicy::Multiple;
         static const AZ::EBusAddressPolicy AddressPolicy = AZ::EBusAddressPolicy::Single;
 
-        // Gives handlers the opportunity to block the loadlevel command.
-        // Return true to stop loading.
+        //! Gives handlers the opportunity to block the loadlevel command.
+        //! There can be multiple handlers for this bus; if any one of them return true the loadlevel command will be stopped.
+        //! @return Return true to stop loading.
         virtual bool ShouldBlockLevelLoading([[maybe_unused]]const char* levelName) { return false; }
     };
+    using LevelLoadBlockerBus = AZ::EBus<LevelLoadBlockerRequests>;
+
+    class LevelSystemLifecycleRequests
+        : public AZ::EBusTraits
+    {
+    public:
+        static const AZ::EBusHandlerPolicy HandlerPolicy = AZ::EBusHandlerPolicy::Single;
+        static const AZ::EBusAddressPolicy AddressPolicy = AZ::EBusAddressPolicy::Single;
+
+        //! Returns the name of the currently loaded level.
+        //! Note: for spawnable level system, this is the cache folder path to the level asset. Example: levels/mylevel/mylevel.spawnable
+        //! @return Level name or empty string if no level loaded.
+        virtual AZStd::string GetCurrentLevelName() = 0;
```

Comment:
This function should be const.
But honestly, this shouldn't even be an EBus.

It should be an interface that gets used with AZ::Interface.

Also this function can return a `const char*`
No need for a `string` at this moment.

```
class ILevelSystemLifecycle
{
    virtual const char* GetCurrentLevelName() const = 0;
};

using LevelSystemLifecycleInterface = AZ::Interface<ILevelSystemLifecycle>;
```

Reply:
Much better :D
Side question: is there still rule against using const EBus functions? I recall that being a thing even for simple stuff like Getters().

Reply:
No, that paradigm of avoiding using const EBus functions was never built on a solid reason.

The thought process by those recommending it was what "if the handler function need to modify the class, but the interface imposed a const function it".

It is not a strong reason why functions shouldn't be able to marked const

Reply:
Thank you. I'm a const fan.